yt/frontends/gadget/io.py

```diff
@@ -77,13 +80,58 @@ def _yield_coordinates(self, data_file, needed_ptype=None):
             yield key, pos
         f.close()
 
+    def _generate_smoothing_length(self, data_files, kdtree):
+        if not self.ds.gen_hsmls:
+            return
+        # TODO: We need to do something like this here but I'm not sure how
+        """
+        if os.path.exists(self.hsml_filename):
+            with open(self.hsml_filename, 'rb') as f:
+                file_hash = struct.unpack('q', f.read(struct.calcsize('q')))[0]
+            if file_hash != self.ds._file_hash:
+                os.remove(self.hsml_filename)
+            else:
+                return
```

Comment:
better ask for forgiveness than permission in Python ([EAFP](https://docs.python.org/3/glossary.html?highlight=ask%20forgiveness))
```python
try:
     with open(self.hsml_filename, 'rb') as f:
         file_hash = struct.unpack('q', f.read(struct.calcsize('q')))[0]

     if file_hash == self.ds._file_hash:
         return

     os.remove(self.hsml_filename)
except FileNotFoundError:
    pass
```

Reply:
See comment below, this needs to be changed entirely.